core/src/main/scala/org/apache/spark/util/collection/Spillable.scala

```diff
@@ -112,7 +112,6 @@ private[spark] abstract class Spillable[C](taskMemoryManager: TaskMemoryManager)
       if (!isSpilled) {
         0L
       } else {
-        _elementsRead = 0
```

Comment:
I delete this code in order to _elementsRead is called by one thread. So we do not make _elementsRead  volatile.